File: 1725. Number Of Rectangles That Can Form The Largest Square.rb
#* 1725. Number Of Rectangles That Can Form The Largest Square
rectangles = [[5,8],[3,9],[5,12],[16,5]]
# @param {Integer[][]} rectangles
# @return {Integer}
def count_good_rectangles(rectangles)
  rectangles.map(&:min).count(rectangles.map(&:min).max)
end
p count_good_rectangles(rectangles)

def count_good_rectangles(rectangles)
  res=rectangles.map(&:min)
  res.count(res.max)
end
p count_good_rectangles(rectangles)

def count_good_rectangles(rectangles)
  rectangles.map{|rec| rec.min }.count(rectangles.map{|rec| rec.min }.max)
end
p count_good_rectangles(rectangles)

File: 1725. Number Of Rectangles That Can Form The Largest Square.py
#* 1725. Number Of Rectangles That Can Form The Largest Square
rectangles = [[5,8],[3,9],[5,12],[16,5]]
#* SOL 1
def count_good_rectangles(rectangles):
  rectsq = [min(subarray) for subarray in rectangles]
  return rectsq.count(max(rectsq)) 

print(count_good_rectangles(rectangles))
#* SOL 1.1 ONELINER
def count_good_rectangles(rectangles):
  return [min(subarray) for subarray in rectangles].count(max([min(subarray) for subarray in rectangles]))

print(count_good_rectangles(rectangles))

print("tests")
#*TEST
for subarray in rectangles:
  print(min(subarray))

rectsq = [min(subarray) for subarray in rectangles]
print(rectsq.count(max([min(subarray) for subarray in rectangles]))) 

rectsq = [min(subarray) for subarray in rectangles]
print(rectsq.count(max(rectsq))) 

#* RUBY TRANSLATION
# def count_good_rectangles(rectangles)
#   rectangles.map{|rec| rec.min }.count(rectangles.map{|rec| rec.min }.max)
# end



